tempesta_fw/ss_skb.c

```diff
@@ -313,8 +368,17 @@ __split_linear_data(struct sk_buff *skb, char *pspt, int len, TfwStr *it)
 	 */
 	if (unlikely((len < 0) && (tail_len == -len))) {
 		ss_skb_put(skb, len);
-		if (skb_is_nonlinear(skb))
+		if (skb_is_nonlinear(skb)) {
 			it->ptr = skb_frag_address(&skb_shinfo(skb)->frags[0]);
+			it->skb = pskb ? pskb : skb;
+		} else {
+			struct sk_buff *dskb = __next_skb_frag(skb, pskb);
+			if (dskb)
+				it->skb = pskb ? pskb : skb;
```

Comment:
The change is very strange. ```pskb``` meand "parent skb", while ```it->skb``` is actullay ```skb``` owning data where ```it->ptr``` points to. Meantime you always write top level (parent) ```skb``` to ```it->skb```.